src/Workspaces/Core/Portable/FindSymbols/Declarations/DeclarationFinder_SourceDeclarations.cs

```diff
@@ -159,6 +205,81 @@ internal static partial class DeclarationFinder
             return list.ToImmutableAndFree();
         }
 
+        internal static async Task<ImmutableArray<SymbolAndProjectId>> FindSourceDeclarationsWithPatternInCurrentProcessAsync(
+            Project project, string pattern, SymbolFilter criteria, CancellationToken cancellationToken)
+        {
+            // The compiler API only supports a predicate which is given a symbol's name.  Because
+            // we only have the name, and nothing else, we need to check it against the last segment
+            // of the pattern.  i.e. if the pattern is 'Console.WL' and we are given 'WriteLine', then
+            // we don't want to check the whole pattern against it (as it will clearly fail), instead
+            // we only want to check the 'WL' portion.  Then, after we get all the candidate symbols
```

Comment:
Does this mean we don't support matching something like `Con.WL`?  (Just curious, not saying we have too).